juju/juju
# Split the cross model facade into a controller facade and model facade

## Description of change

This PR is 99% moving around existing code.

The "CrossModelRelations" facade is split into 2 facades:
1. ApplicationOffers, a model facade
2. RemoteEndpoints, a controller facade

The ApplicationOffers facade is for managing offers in a model. It currently has Offer() and List() and will get Remove(), Update() etc

The RemoteEndpoints facade is for querying info about offered endpoints in any model on a controller.

## QA steps

Run up a cross model relations deployment and check that the existing cross model CLI all works.
